src/en/kunmangato/src/eu/kanade/tachiyomi/extension/en/kunmangato/KunMangaTo.kt

```diff
@@ -0,0 +1,267 @@
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+import kotlin.concurrent.thread
+
+class KunMangaTo : ParsedHttpSource() {
+    override val name = "KunMangaTo"
+    override val baseUrl = "https://kunmanga.to"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("MMM d, yyyy", Locale.ENGLISH)
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET(baseUrl)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(popularMangaSelector()).map { element ->
+            popularMangaFromElement(element)
+        }.associateBy({ it.url }, { it }).values.toList()
+        return MangasPage(mangas, false)
+    }
+
+    override fun popularMangaSelector(): String {
+        return ".sidebar-box-popular article"
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        val a = element.select("a.manga").first()!!
+        return SManga.create().apply {
+            title = a.text()
+            url = a.attr("href").removePrefix(baseUrl)
+            thumbnail_url = element.select("img").first()!!.attr("src")
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val urlBuilder = "$baseUrl/search".toHttpUrl().newBuilder()
+            .addQueryParameter("term", query.trim())
+            .addQueryParameter("page", page.toString())
+
+        filters.filterIsInstance<UriPartFilter>().forEach { filter ->
+            urlBuilder.addQueryParameter(
+                filter.internalName,
+                filter.toUriPart(),
+            )
+        }
+
+        return GET(urlBuilder.build())
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(searchMangaSelector()).map { element ->
+            searchMangaFromElement(element)
+        }
+        val nextPageElement = document.select(".pagination .page-item").last()
+        return MangasPage(
+            mangas,
+            nextPageElement?.hasClass("disabled")?.not() ?: false,
+        )
+    }
+
+    override fun searchMangaSelector(): String {
+        return "article .card-manga"
+    }
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val urlBuilder = "$baseUrl/latest".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        return GET(urlBuilder.build())
```

Comment:
send headers here too

Reply:
Are you sure? Headers are only needed for "/chapter-resources", because that's an API endpoint, that returns JSON and is called by the website after manga's page is loaded. Every other request returns static HTML page

Reply:
Any connection needs headers, in this case the default headers only contain the User-Agent, necessary to deal with problems like cloudflare

Reply:
Sorry, I misunderstood you initially. I thought you mean csrf token and cookies headers. Never mind my question. Will fix